### PR TITLE
Integration tests for tpp technical app and financroo

### PR DESCRIPTION
**Pull request by mzerko-cloudentity**

### CODE REVIEW

File: tests/cypress/integration/FinancrooAppTests.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  [
+    [billsAccount, householdAccount],
+    [billsAccount],
+    [householdAccount],
+    []
+  ].forEach(accounts => {
+    it(`Happy path with accounts: ${accounts}`, () => {
+      financrooLoginPage.visit()
+      financrooLoginPage.login()
+      acpLoginPage.login(`test`, `p@ssw0rd!`)
```
Comment (plandzwojczak-cloudentity) [inline tests/cypress/integration/FinancrooAppTests.ts:30]: Can we create such const file for contant like BASE_TEST_PASSWORD or sth like that?

File: tests/cypress/integration/FinancrooAppTests.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import {AcpLoginPage} from '../pages/acp/AcpLoginPage';
+import {FinancrooLoginPage} from '../pages/financroo/FinancrooLoginPage';
+import {ConsentPage} from '../pages/consent/ConsentPage';
+import {ErrorPage} from '../pages/ErrorPage';
+import {FinancrooWelcomePage} from '../pages/financroo/FinancrooWelcomePage';
+import {FinancrooConnectAccountPage} from '../pages/financroo/FinancrooConnectAccountPage';
+import {FinancrooDashboardPage} from '../pages/financroo/FinancrooDashboardPage';
+
+describe(`Financroo app`, () => {
```
Comment (plandzwojczak-cloudentity) [inline tests/cypress/integration/FinancrooAppTests.ts:9]: If I understand correctly right now there aren't any REST preconditions like creating tenant/authorization server etc (don't know what's exactly here but I suppose you got my point). Did you think about it how it should be handled? Some abstractions (helpers, managers?) dedicated only for that purpose?

File: .github/workflows/cypress.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: cypress
```
Comment (plandzwojczak-cloudentity) [inline .github/workflows/cypress.yml:1]: Is there any README file for cypress. I mean env setup locally + how to run whole test suite/ single test from terminal?

Comment (plandzwojczak-cloudentity) [reply .github/workflows/cypress.yml:1]: Or just references :)

File: tests/cypress/integration/FinancrooAppTests.ts (hunk 1 above)
Comment (mzerko-cloudentity) [reply tests/cypress/integration/FinancrooAppTests.ts]: Done

File: .github/workflows/cypress.yml (hunk 3 above)
Comment (mzerko-cloudentity) [reply .github/workflows/cypress.yml:1]: Done

File: tests/cypress/integration/FinancrooAppTests.ts (hunk 2 above)
Comment (mzerko-cloudentity) [reply tests/cypress/integration/FinancrooAppTests.ts:10]: I guess there need to be some helpers to do so. But so far I've only needed to do one rest call

**Review by mbilski — APPROVED** [review]

File: tests/cypress/integration/FinancrooAppTests.ts (hunk 2 above)
Comment (plandzwojczak-cloudentity) [reply tests/cypress/integration/FinancrooAppTests.ts:10]: so at this moment, it's ok, but we have to take it into consideration in the future :)

**Comment by plandzwojczak-cloudentity** [comment]
1. Do we have some report tool (mocha/allure?) to easily navigate with cypress results? If yes could you provide some screenshot? Does it suit us?

2. Open question - I saw that with cypress there is a way to go with the FeatureFiles approach. I mean more bdd/gherkin/cucumber syntax. Would we like to implement such approach?  For sure it will make it easier reading the scenarios, but still this is cucumber :smile:

File: tests/cypress/integration/FinancrooAppTests.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      acpLoginPage.login(Credentials.tppUsername, Credentials.defaultPassword)
+      consentPage.checkAccounts(accounts)
+      consentPage.assertPermissions([`ReadAccountsDetail`, `ReadAccountsBasic`, `ReadBalances`,
+        `ReadTransactionsBasic`, `ReadTransactionsDetail`, `ReadTransactionsCredits`, `ReadTransactionsDebits`])
+      consentPage.confirm()
+      financrooDashboardPage.assertAccounts(accounts)
+    })
+  })
+
+  it(`Cancel on ACP login`, () => {
```
Comment (plandzwojczak-cloudentity) [inline tests/cypress/integration/FinancrooAppTests.ts:45]: This my suggestion, but shouldn't we have more descriptive test scenarios name? (I assume that "it" it's a test scenario and "describe" its Test class name comparing to java world). For example
- should Rejected authentication Error Be Displayed when X User cancel Login to Acp
- As x User Cancel Login to ACP
What do you think?

Comment (plandzwojczak-cloudentity) [reply tests/cypress/integration/FinancrooAppTests.ts:45]: Rather than happy path etc

File: .github/workflows/cypress.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+on:
+  push:
+  pull_request:
```
Comment (holowinski) [inline .github/workflows/cypress.yml:5]: Is the PR opened necessary? Isn't this duplicating push?

**Review by holowinski — APPROVED** [review]

**Review by ikawalec — APPROVED** [review]

File: .github/workflows/cypress.yml (hunk 5 above)
Comment (mzerko-cloudentity) [reply .github/workflows/cypress.yml]: Removed

File: .github/workflows/cypress.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    types: [ opened ]
```
[the remark this answers is not in the document]
Comment (mzerko-cloudentity) [reply .github/workflows/cypress.yml]: Should be okay

**Pull request merged**